crypto/bn/bn_mont.c

```diff
@@ -95,8 +95,10 @@ static int BN_from_montgomery_word(BIGNUM *ret, BIGNUM *r, BN_MONT_CTX *mont)
 
     /* clear the top words of T */
     i = max - r->top;
-    if (i)
+    if (i > 0)
         memset(&rp[r->top], 0, sizeof(*rp) * i);
+    else if (i < 0)
```

Comment:
If you re-ordered the statements, then you avoid the else.

Reply:
do you mean
```
if (i<0)
   return 0;
if (i)
  memset(....)
```

Reply:
Yes, exactly that.  Assuming you do so, +1/approved from me.

Reply:
okay, fine.